app/scripts/modules/core/src/cancelModal/CancelModal.tsx

```diff
@@ -0,0 +1,128 @@
+import * as React from 'react';
+import { Modal } from 'react-bootstrap';
+import { Field, FieldProps, Form, Formik } from 'formik';
+
+import { SubmitButton, ModalClose } from 'core';
+import { ReactModal } from 'core/presentation';
+
+export interface ICancelModalProps {
+  body?: string;
+  buttonText: string;
+  cancelButtonText?: string;
+  header: string;
+  submitMethod: (reason: string, force?: boolean) => ng.IPromise<any>;
+  closeModal?(result?: any): void; // provided by ReactModal
+  dismissModal?(rejection?: any): void; // provided by ReactModal
+}
+
+export interface ICancelModalState {
+  isSubmitting: boolean;
+  error: boolean;
+  errorMessage?: string;
+}
+
+export interface ICancelModalValues {
+  reason?: string;
+}
+
+export class CancelModal extends React.Component<ICancelModalProps, ICancelModalState> {
+  public static confirm(props: ICancelModalProps): Promise<void> {
+    return ReactModal.show(CancelModal, props);
+  }
+
+  constructor(props: ICancelModalProps) {
+    super(props);
+
+    this.state = {
+      isSubmitting: false,
+      error: false,
+    };
+  }
+
+  private close = (args?: any): void => {
+    this.props.dismissModal.apply(null, args);
+  };
+
+  private showError = (exception: string): void => {
+    this.setState({
+      error: true,
+      errorMessage: exception,
+      isSubmitting: false,
+    });
+  };
+
+  private submitConfirmation = (values: ICancelModalValues): void => {
+    this.setState({ isSubmitting: true });
+    this.props.submitMethod(values.reason).then(this.close, this.showError);
+  };
+
+  public render() {
+    const { header, body, buttonText, cancelButtonText = 'Cancel' } = this.props;
```

Comment:
Not used to the pattern of defaults here.

I typically like to see the `public defaultProps: Partial<ICancelModalProps> = { cancelButtonText: 'Cancel' }` pattern, just in case in the future we expect `cancelButtonText` to exist elsewhere than than render function.

Reply:
Ah right, default props. Thanks!